### ISSUE TITLE
Duplicated `$ref` When Defining Properties in Traits

### ISSUE THREAD

**Issue by adrian-enspired** [zircote/swagger-php#712]
I have a schema defined on a class, similar to:
```
/**
 * @OA\Schema(schema="foo")
 */
class Foo {
  //  . . .
}
```
I have a property on a trait which references this, using `allOf`:
```
trait Fooer {
  /**
   * Some Foo.
   *
   * @var Foo
   * @OA\Property(
   *  allOf={
   *   @OA\Schema(title="Some Foo"),
   *   @OA\Schema(ref="#/components/schemas/foo")
   *  }
   * )
   */
  public $a_foo;
}

class FooUser {
  use Fooer;
}
```
This results in an invalid spec being generated, with an extra `$ref` in the property:
```
a_foo:
  allOf:
    -
      title: 'Some Foo'
    -
      $ref: '#/components/schemas/foo'
  $ref: '#/components/schemas/foo'
```
in addition, the description is omitted.

**If** I make the exact same property and annotations on the concrete class, the generated spec looks as expected (and is valid):
```
a_foo:
  description: 'Some Foo.'
  allOf:
    -
      title: 'Some Foo'
    -
      $ref: '#/components/schemas/foo'
```
this same result can be produced by removing the `@var Foo` annotation from the `$a_foo` property, so I assume php-swagger is seeing the schema definition on the class `Foo` in addition to the ref on the trait, and failing to de-dupe them. However, I'm not sure this would explain why it generates a malformed spec.

**Comment by DerManoMann** [comment]
Here is a new [PR](https://github.com/zircote/swagger-php/pull/774) to improve handling of traits. Feel free to try that and provide some feedback on the PR.

**Comment by DerManoMann** [comment]
Fixed via #818, #804, #798, #782, #779
If you still see problems please raise a new issue. Thanks.
